### PR TITLE
[Merged by Bors] - feat(ModelTheory/Basic): additional simp lemmas

### PR DESCRIPTION
**Pull request by GabinKolly**
Also results about injectivity.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/ModelTheory/ElementaryMaps.lean [hunk 1]
```diff
@@ -86,12 +86,12 @@ theorem map_boundedFormula (f : M ↪ₑ[L] N) {α : Type*} {n : ℕ} (φ : L.Bo
         (Sum.elim (v ∘ (↑)) xs ∘ (Fintype.equivFin _).symm)
     simp only [Formula.realize_relabel, BoundedFormula.realize_toFormula, iff_eq_eq] at h
     rw [← Function.comp.assoc _ _ (Fintype.equivFin _).symm,
-      Function.comp.assoc _ (Fintype.equivFin _).symm (Fintype.equivFin _), Equiv.symm_comp_self,
-      Function.comp_id, Function.comp.assoc, Sum.elim_comp_inl,
+      Function.comp.assoc _ (Fintype.equivFin _).symm (Fintype.equivFin _),
+      _root_.Equiv.symm_comp_self, Function.comp_id, Function.comp.assoc, Sum.elim_comp_inl,
```
Comment (ocfnash) [inline Mathlib/ModelTheory/ElementaryMaps.lean:90]: Another way to disambiguate is to use the keyword `protected` (as in `protected lemma symm_comp_self`) when defining the new lemmas. This tells Lean that an references must fully qualify the name.

**Review by ocfnash — APPROVED** [review]
I recommend adding the keyword `protected` to those lemmas whose name also exists in a shallower namespace, but I do not insist.

Thanks!

bors d+

**Comment by QuinnLesquimau** [comment]
bors d+

**Pull request closed without merging**